### PR TITLE
Add objective in json checklist - close #589

### LINKED ISSUE #589: Add "Objectives": "<Grab objectives>" in JSON checklist
Need to update the current JSON checklist with `Objectives`

- Objective can be updated once we have an objective section in each test case.

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR covers issue #589 
- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Add objective in JSON checklist

### CODE REVIEW

**Comment by ThunderSon** [comment]
Can we try and handle the objectives as an array?
It's always an `Objectives` object that has an UL.
Can we ignore as well sub-tests? So like SQLi for Oracle. Those are sub-tests that don't have an ID, so we can scratch on that?
@kingthorin that is why I wanted to discuss the old/removed IDs, to help with filtering in here.

@rejahrehim I am looking into `jq` command line, instead of brute forcing what we're doing.
We can use it to format the document when everything is done.
We can use it as well to add the objects, instead of bashing everything :p
I noticed as well that the CLNT category's name is `Clien`, so that wasn't parsed properly.

Do you think we need to add a certain value to deprecated tests so that we can put them in the checklist as "deprecated", but still mention them? Then we'd parse on that value.

**Comment by rejahrehim** [comment]
@ThunderSon  If `Objectives` always a list we can put that in an array. I will update the PR.
Also, Let me try removing the sub-tests.

**Comment by rejahrehim** [comment]
I haven't added `jq` as it will add some more dependency in build. If fine we can use `jq`

**Comment by kingthorin** [comment]
The stable deploy job uses yq (which is similar) and a single snap install command.

**Comment by ThunderSon** [comment]
Haha please use `jq`. That's a dependency I'm willing to allow for the massacre we're doing :p
Yes the objectives are always going to be a list.

**Comment by rejahrehim** [comment]
@ThunderSon @kingthorin  Fixed the issues without `jq`. I will update the script with `jq`. Can we handle that in another PR?

**Comment by ThunderSon** [comment]
Reviewing the generated checklist.
Of course it can be another PR to keep things separated. Thought of it as it might have made things easier, seems you're good without it :)

File: .github/scripts/make-json.sh [hunk 1]
```diff
@@ -31,23 +39,54 @@ for d in */ ; do
...
+
+                        # Get test name and reference link from the file
+                        read -r tname < $file
+                        tname=${tname:2}
+                        tref=`echo $file | sed 's/.md/.html/'`
+                        # Add test ID, test name and reference link from the file
+                        echo "                \"name\":\"${tname}\","  >> checklist.json
+                        echo "                \"id\":\"${tid}\","  >> checklist.json
+                        echo "                \"Reference\":\"https://owasp.org/www-project-web-security-testing-guide/stable/4-Web_Application_Security_Testing/$tref\","  >> checklist.json
+                        echo "                \"Objective\":["  >> checklist.json
```
Comment (kingthorin) [inline .github/scripts/make-json.sh:71]: Do you think it makes anyone's lives easier if the element names are all lower case? (Maybe I'm over thinking it)

Comment (rejahrehim) [reply .github/scripts/make-json.sh:71]: I think, better to have all lower case. let me update that too

Comment (rejahrehim) [reply .github/scripts/make-json.sh:71]: `objective` or `objectives` ?

Comment (kingthorin) [reply .github/scripts/make-json.sh:71]: Lets keep the plural S since it's an array.

**Review by kingthorin — APPROVED** [review]
Thanks!

**Review by ThunderSon — APPROVED** [review]
Thanks! Looks lovely. Fixed a small 2 layered UL that wasn't cute in the parsing. Otherwise everything looks good.

**Pull request merged**